docs/settings/python_analysis_includeAliasesFromUserFiles.md

```diff
@@ -0,0 +1,157 @@
+
+## The `python.analysis.includeAliasesFromUserFiles` Setting
+
+The `python.analysis.includeAliasesFromUserFiles` setting controls if alias symbols from user files are included in auto-import and add import suggestions.
+
+- `true`: Includes alias symbols from user files.
+- `false` (default): Only suggests symbols from their original definitions.
+
+## How to Change the Setting
+
+1. **Open Settings**: Click on the gear icon in the lower-left corner of Visual Studio Code and select **Settings**.
+2. **Search for the Setting**: Type `python.analysis.includeAliasesFromUserFiles` in the search bar.
+3. **Modify the Setting**: Set it to `true` or `false` as needed.
+
+Alternatively, you can edit your `settings.json` file directly:
+
+1. **Open Settings (JSON)**: Open the command palette and type `Preferences: Open Settings (JSON)`.
+2. **Add or Modify the Setting**:
+   ```json
+   "python.analysis.includeAliasesFromUserFiles": true
+   ```
+
+## When and Why to Enable `includeAliasesFromUserFiles`
+
+### Improved Auto-Import Suggestions
+
+By enabling `python.analysis.includeAliasesFromUserFiles`, Pylance will include alias symbols from your user files in features like auto-import suggestions. This can be particularly useful when you have organized your codebase to expose certain symbols through a central module or when you frequently use aliases to simplify imports.
+
+Consider the following scenario with the following project structure:
+
+```
+my_project/
+├── __init__.py
+├── utils.py
+├── nested/
+│   ├── redundant.py
+│   ├── all.py
+└── usage.py
+```
+
+**`utils.py`**
+
+```python
+def helper_function():
+    print("This is the helper function")
+```
+
+**`__init__.py`**
+
+```python
+from .utils import helper_function
+```
+
+**`nested/redundant.py`**
+
+```python
+from ..utils import helper_function as helper_function
+```
+
+**`nested/all.py`**
+
+```python
+from my_project.utils import helper_function
+__all__ = ["helper_function"]
+```
+
+**`usage.py`**
+
+```python
+from .utils import helper_function
+```
+
+When using auto-import, Pylance will choose the best alias symbol based on the following criteria, with each criterion contributing 1 point:
+
+- The alias symbol is in an `__init__.py` file.
+- The alias symbol is in a redundant form (e.g., `import helper_function as helper_function`).
+- The alias symbol is listed in `__all__`.
+
+The alias symbol with the highest score will be chosen. In case of a tie, the symbol with the shortest import module path will be selected.
+
+For example, the alias symbol from `usage.py` has the lowest score and will be disregarded. The alias symbols from `__init__.py`, `redundant.py`, and `all.py` each have 1 point, but the one from `__init__.py` has the shortest import module path, so it will be selected. If `__init__.py` had used `from .utils import helper_function as helper_function`, it would receive 2 points and be selected.
```

Comment:
```suggestion
For example, the alias symbol from `usage.py` has the lowest score and will be disregarded. The alias symbols from `__init__.py`, `redundant.py`, and `all.py` each have 1 point, but the one from `__init__.py` has the shortest import module path, so it will be selected. If `__init__.py` had used `from .utils import helper_function as helper_function` instead, it would have received 2 points and be selected.
```